outport/process/factory/check_test.go

```diff
@@ -0,0 +1,78 @@
+package factory
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/outport/process"
+	"github.com/ElrondNetwork/elrond-go/outport/process/alteredaccounts"
+	"github.com/ElrondNetwork/elrond-go/outport/process/transactionsfee"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/testscommon/economicsmocks"
+	"github.com/ElrondNetwork/elrond-go/testscommon/genericMocks"
+	"github.com/ElrondNetwork/elrond-go/testscommon/shardingMocks"
+	"github.com/ElrondNetwork/elrond-go/testscommon/state"
+	"github.com/stretchr/testify/require"
+)
+
+func createArgCreateOutportDataProvider() ArgOutportDataProviderFactory {
+	return ArgOutportDataProviderFactory{
+		HasDrivers:             false,
+		AddressConverter:       &testscommon.PubkeyConverterMock{},
+		AccountsDB:             &state.AccountsStub{},
+		Marshaller:             &testscommon.MarshalizerMock{},
+		EsdtDataStorageHandler: &testscommon.EsdtStorageHandlerStub{},
+		TransactionsStorer:     &genericMocks.StorerMock{},
+		ShardCoordinator:       &testscommon.ShardsCoordinatorMock{},
+		TxCoordinator:          &testscommon.TransactionCoordinatorMock{},
+		NodesCoordinator:       &shardingMocks.NodesCoordinatorMock{},
+		GasConsumedProvider:    &testscommon.GasHandlerStub{},
+		EconomicsData:          &economicsmocks.EconomicsHandlerMock{},
+	}
+}
+
+func TestCheckArgCreateOutportDataProvider(t *testing.T) {
+	t.Parallel()
+
+	arg := createArgCreateOutportDataProvider()
+	arg.AddressConverter = nil
+	require.Equal(t, alteredaccounts.ErrNilPubKeyConverter, checkArgCreateOutportDataProvider(arg))
```

Comment:
perhaps rename the original method to `checkArgOutportDataProviderFactory` as well

Reply:
done